### ISSUE TITLE
[feature request] render template directly from string rather than from a file

### ISSUE THREAD

**Issue by iptq** [SergioBenitez/Rocket#462]
The `handlebars-rust` and `tera` libs both have APIs that allow rendering HTML directly from a template string rather than through providing a file name. Rocket's Template class should mirror this as well. The changes necessary seem pretty involved; right now the Template struct contains the name of the file + the context. I think rather than holding the name of the file, it should hold the actual template string. Then you could have something like `Template::render_string` or `Template::render` (from file) which basically just does the same thing as render_string except it reads it from a file.

I'm not sure how *often* this use case would come up, but it seems like a feature that would be nice to have (I also need it in one of my projects!)

**Comment by SergioBenitez** [comment]
The `Template` type aims to be as agnostic as possible about which underlying templating engine is used. However, for this feature, we'd need the user to inform us about which templating engine to use when rendering the string. What's more, in order to escape input properly, the user would also need to explicitly specify the template's file type. As a result, any `render_string` method invocation would look something like:

```rust
Template::render_string("my {{ string }}", context, Engine::Tera, FileType::Html);
```

This API seems rather unergonomic. We might imagine improving ergonomics with a macro of some sort:

```rust
Template::render_string("my {{ string }}", context, ext!(".tera.html"));
```

But now, this seems pretty magical.

In short, adding this feature would mean introducing a complicated, error-prone API. For this reason alone, I'm not inclined to accept this request. As a secondary reason, template rendering performance would suffer as the string would need be parsed as a template on the fly.

What's your use case here? By the way, you can always use `tera` or `handlebars` directly.

**Comment by SergioBenitez** [comment]
#873 was merged here which proposes that Rocket allow a string to be registered as a template.

To be clear, _this_ issue requests that Rocket allow a string to be used in place of a registered template in an ad-hoc fashion, just prior to returning a `Template` from a handler. #873, on the other hand, requests that a string be allowed to be registered as a template, presumably via [`Template::custom`](https://api.rocket.rs/v0.4/rocket_contrib/templates/struct.Template.html#method.custom) (which Rocket currently ignores), and then used in a named fashion as file-based templates are today. Note that #873 subsumes the request in this issue when the template string is known apriori but not otherwise.

**Comment by SergioBenitez** [comment]
This is now possible by registering a template directly with the engine. See #1637.
